Installation issue: openmpi 4.1.5: prrte version constraint is required

### Steps to reproduce the issue

```console
# spack spec -I openmpi@4.1.5 %oneapi@2025.2.0
...
 -   openmpi@4.1.5+atomics~cuda~cxx~cxx_exceptions~debug+fortran~gpfs~internal-hwloc~internal-libevent~internal-pmix~ipv6~java~lustre~memchecker~openshmem~orterunprefix~rocm+romio+rsh~singularity~static~two_level_namespace+vt+wrapper-rpath build_system=autotools fabrics:=none romio-filesystem:=none schedulers:=none arch=linux-rocky8-x86_64_v3 %c,cxx,fortran=oneapi@2025.2.0
[+]      ^compiler-wrapper@1.0 build_system=generic arch=linux-rocky8-x86_64_v3 
[e]      ^glibc@2.28 build_system=autotools arch=linux-rocky8-x86_64_v3 
[+]      ^gmake@4.4.1~guile build_system=generic arch=linux-rocky8-x86_64_v3 %c=gcc@8.5.0
[e]          ^gcc@8.5.0~binutils+bootstrap~graphite~nvptx~piclibs~profiled~strip build_system=autotools build_type=RelWithDebInfo languages:='c,c++,fortran' patches:=98a9c96,d4919d6 arch=linux-rocky8-x86_64_v3 
[+]          ^gcc-runtime@8.5.0 build_system=generic arch=linux-rocky8-x86_64_v3 
[+]      ^hwloc@2.11.1~cairo~cuda~gl~level_zero~libudev+libxml2~nvml~opencl+pci~rocm build_system=autotools libs:=shared,static arch=linux-rocky8-x86_64_v3 %c,cxx=gcc@8.5.0
[+]          ^libpciaccess@0.17 build_system=autotools arch=linux-rocky8-x86_64_v3 %c=gcc@8.5.0
[+]              ^util-macros@1.20.1 build_system=autotools arch=linux-rocky8-x86_64_v3 
[+]          ^libxml2@2.13.5~http+pic~python+shared build_system=autotools arch=linux-rocky8-x86_64_v3 %c=gcc@8.5.0
[+]              ^libiconv@1.18 build_system=autotools libs:=shared,static arch=linux-rocky8-x86_64_v3 %c=gcc@8.5.0
[+]              ^xz@5.6.3~pic build_system=autotools libs:=shared,static arch=linux-rocky8-x86_64_v3 %c=gcc@8.5.0
[+]          ^ncurses@6.5~symlinks+termlib abi=none build_system=autotools patches:=7a351bc arch=linux-rocky8-x86_64_v3 %c,cxx=gcc@8.5.0
[e]      ^intel-oneapi-compilers@2025.2.0~amd+envmods~nvidia build_system=generic arch=linux-rocky8-x86_64 
[+]      ^intel-oneapi-runtime@2025.2.0 build_system=generic arch=linux-rocky8-x86_64_v3 
[+]      ^libevent@2.1.12+openssl build_system=autotools arch=linux-rocky8-x86_64_v3 %c=gcc@8.5.0
[+]          ^openssl@3.4.1~docs+shared build_system=generic certs=mozilla arch=linux-rocky8-x86_64_v3 %c,cxx=gcc@8.5.0
[+]              ^ca-certificates-mozilla@2025-05-20 build_system=generic arch=linux-rocky8-x86_64_v3 
[+]      ^numactl@2.0.18 build_system=autotools arch=linux-rocky8-x86_64_v3 %c=gcc@8.5.0
[+]          ^autoconf@2.72 build_system=autotools arch=linux-rocky8-x86_64_v3 
[+]          ^automake@1.16.5 build_system=autotools arch=linux-rocky8-x86_64_v3 %c=gcc@8.5.0
[+]          ^libtool@2.4.6 build_system=autotools arch=linux-rocky8-x86_64_v3 %c=gcc@8.5.0
[+]              ^findutils@4.10.0 build_system=autotools patches:=440b954 arch=linux-rocky8-x86_64_v3 %c=gcc@8.5.0
[+]          ^m4@1.4.20+sigsegv build_system=autotools arch=linux-rocky8-x86_64_v3 %c,cxx=gcc@8.5.0
[+]              ^diffutils@3.10 build_system=autotools arch=linux-rocky8-x86_64_v3 %c=gcc@8.5.0
[+]              ^libsigsegv@2.14 build_system=autotools arch=linux-rocky8-x86_64_v3 %c=gcc@8.5.0
[+]      ^openssh@9.9p1+gssapi build_system=autotools arch=linux-rocky8-x86_64_v3 %c,cxx=gcc@8.5.0
[+]          ^krb5@1.21.3+shared build_system=autotools arch=linux-rocky8-x86_64_v3 %c,cxx=gcc@8.5.0
[+]              ^bison@3.8.2~color build_system=autotools arch=linux-rocky8-x86_64_v3 %c,cxx=gcc@8.5.0
[+]              ^gettext@0.23.1+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-rocky8-x86_64_v3 %c,cxx=gcc@8.5.0
[+]                  ^bzip2@1.0.8~debug~pic+shared build_system=generic arch=linux-rocky8-x86_64_v3 %c=gcc@8.5.0
[+]                  ^tar@1.35 build_system=autotools zip=pigz arch=linux-rocky8-x86_64_v3 %c=gcc@8.5.0
[+]                      ^pigz@2.8 build_system=makefile arch=linux-rocky8-x86_64_v3 %c=gcc@8.5.0
[+]                      ^zstd@1.5.7+programs build_system=makefile compression:=none libs:=shared,static arch=linux-rocky8-x86_64_v3 %c,cxx=gcc@8.5.0
[+]          ^libedit@3.1-20240808 build_system=autotools arch=linux-rocky8-x86_64_v3 %c=gcc@8.5.0
[+]          ^libxcrypt@4.4.38~obsolete_api build_system=autotools arch=linux-rocky8-x86_64_v3 %c=gcc@8.5.0
[e]      ^perl@5.26.3+cpanm+opcode+open+shared+threads build_system=generic patches:=8cf4302 arch=linux-rocky8-x86_64_v3 
[+]      ^pkgconf@2.5.1 build_system=autotools arch=linux-rocky8-x86_64_v3 %c=gcc@8.5.0
[+]      ^pmix@4.2.2~munge~python~restful build_system=autotools arch=linux-rocky8-x86_64_v3 %c=gcc@8.5.0
 -       ^prrte@4.0.0 build_system=autotools arch=linux-rocky8-x86_64_v3 %c=oneapi@2025.2.0
[+]          ^flex@2.6.3+lex~nls build_system=autotools arch=linux-rocky8-x86_64_v3 %c,cxx=gcc@8.5.0
[+]      ^zlib-ng@2.2.4+compat+new_strategies+opt+pic+shared build_system=autotools arch=linux-rocky8-x86_64_v3 %c,cxx=gcc@8.5.0
```

### Error message

<details><summary>Error message</summary>

<pre>
$ spack install openmpi@4.1.5 %oneapi@2025.2.0
...
==> Installing prrte-4.0.0-rzke5aegotye6i3ttlebpgmd4hcxjnbq [35/37]
==> Fetching https://github.com/pmix/prrte/releases/download/v4.0.0/prrte-4.0.0.tar.bz2
    [100%]    4.46 MB @   12.6 MB/s
==> No patches needed for prrte
==> prrte: Executing phase: 'autoreconf'
==> prrte: Executing phase: 'configure'
==> Error: ProcessError: Command exited with status 1:
    '/tmp/root/spack-stage/spack-stage-prrte-4.0.0-rzke5aegotye6i3ttlebpgmd4hcxjnbq/spack-src/configure' '--prefix=/opt/release/linux-x86_64_v3/prrte-4.0.0-rzke5aegotye6i3ttlebpgmd4hcxjnbq' '--enable-shared' '--enable-static' '--disable-sphinx' '--with-libevent=/opt/release/linux-x86_64_v3/libevent-2.1.12-ajxv7tvt4665nytvpruygvqxddiz3sbd' '--with-hwloc=/opt/release/linux-x86_64_v3/hwloc-2.11.1-ccu3h42hjqaitkg764mhghxpwvujdilq' '--with-pmix=/opt/release/linux-x86_64_v3/pmix-4.2.2-puawe3cd6gxkiqv4xxgzzzdaflyjmpk2'

1 error found in build log:
     430    checking pmix.h usability... yes
     431    checking pmix.h presence... yes
     432    checking for pmix.h... yes
     433    checking for PMIx_Init... yes
     434    checking version at or above v6.0.0... no
     435    configure: WARNING: PRRTE requires PMIx v0x00060000 or above.
  >> 436    configure: error: Please select a supported version and configure a
            gain
</pre></details>


### Information on your system

* **Spack:** 1.0.0 (https://github.com/spack/spack/commit/73eaea13f381e3495299284856fd02a64e1d154c)
* **Builtin repo:** https://github.com/spack/spack-packages/commit/9aa76d64cd9a4e6c2418e55d2dca93820f3362f7
* **Python:** 3.6.8
* **Platform:** linux-rocky8-skylake


### Additional information

In https://github.com/spack/spack/pull/40686, the following dependency version constraint was added to the openmpi Spack Package Recipe (SPR):
```
        # pmix@4.2.3 contains a breaking change, compat fixed in openmpi@4.1.6
        # See https://www.mail-archive.com/announce@lists.open-mpi.org//msg00158.html
        depends_on("pmix@:4.2.2", when="@:4.1.5")
```

In https://github.com/spack/spack-packages/pull/241 , the following dependency was added without version constraints to the openmpi SPR:
```
        # When an external PMIx is used, also an external PRRTE should be used
        # https://github.com/open-mpi/ompi/issues/13275#issuecomment-2907903468
        depends_on("prrte")
```

There's a version dependency relationship between `pmix` and `prrte` that appears to have been missed. `prrte` 3.0.1 requires `pmix` 4.2.4.

i.e. https://github.com/openpmix/prrte/compare/v3.0.0...v3.0.1 . Specifically: https://github.com/openpmix/prrte/commit/63370ca00771a7a6004d6b638476ca794b04e4c1
```
        # -pmix_min_version=4.1.2
        # +pmix_min_version=4.2.4
```

When openmpi 4.1.5 requires pmix 4.2.2 or older, it is also necessary to require prrte 3.0.0 or older. i.e.
```
        depends_on("pmix@:4.2.2", when="@:4.1.5")
        depends_on("prrte@:3.0.0", when="@:4.1.5")
```

Please note, this issue is orthogonal to the issue that is fixed by the following PRs:
https://github.com/spack/spack-packages/pull/864
https://github.com/spack/spack-packages/pull/917

@hppritcha @naughtont3 

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/repos/spack_repo/builtin/packages/prrte/package.py
+++ b/repos/spack_repo/builtin/packages/prrte/package.py
@@ -46,6 +46,12 @@ class Prrte(AutotoolsPackage):
 
     depends_on("pmix")
     depends_on("pmix@:5", when="@:3")
+    # NOTE: prrte 3.0.1 requires pmix 4.2.4
+    # https://github.com/openpmix/prrte/compare/v3.0.0...v3.0.1
+    # https://github.com/openpmix/prrte/commit/63370ca00771a7a6004d6b638476ca794b04e4c1
+    # -pmix_min_version=4.1.2
+    # +pmix_min_version=4.2.4
+    depends_on("pmix@4.2.4:", when="@3.0.1:")
     depends_on("libevent")
     depends_on("hwloc")
     depends_on("perl", type=("build"))

